clang-tools-extra/clang-tidy/bugprone/UnusedLocalNonTrivialVariableCheck.cpp

```diff
@@ -29,6 +29,12 @@ static constexpr StringRef DefaultIncludeTypeRegex =
 
 AST_MATCHER(VarDecl, isLocalVarDecl) { return Node.isLocalVarDecl(); }
 AST_MATCHER(VarDecl, isReferenced) { return Node.isReferenced(); }
+AST_MATCHER_P(VarDecl, explicitMarkUnused, LangOptions, LangOpts) {
+  // Implementations should not emit a warning that a name-independent
+  // declaration is used or unused.
+  return Node.hasAttr<UnusedAttr>() ||
+         (LangOpts.CPlusPlus26 && Node.isPlaceholderVar(LangOpts));
```

Comment:
Out of curiosity, is the CplusPlus26 needed? Won't `isPlaceholderVar` return `false` otherwise?

Reply:
And, you dont need to pass LangOptions via argument.
From matcher you got access to ASTContext, and that class got [getLangOpts](https://clang.llvm.org/doxygen/classclang_1_1ASTContext.html#a50f31d935973dd09016088579ba81944)

Reply:
> Out of curiosity, is the CplusPlus26 needed? Won't isPlaceholderVar return false otherwise?

Yes, `isPlaceholderVar` does not check cpp version.